### ISSUE TITLE
Maps from a secondary mod cannot use assets from a different mod

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#607]
TMX maps use assets like tilesets, images, etc, from other locations defined by relative paths.  Currently if you use a map from a secondary mod (not tuxemon), but the map uses resources from tuxemon, like images, etc, then the map loading will fail.  In this case the only solution is to edit the map and change the file paths to point to the tuxemon mod folder.

**Comment by xirsoi** [comment]
I feel like a possible solution to this would be a virtual directory structure that would be the result of merging the actual structures of the mods + base game.

This would come with the bonus of secondary mods getting the benefit of, say, file replacements performed by other secondary mods that they don't know about.

I imagine this would be a fair bit of work and probably a source of a lot of regressions.

**Comment by Russell-Tran** [comment]
Is the easiest solution here for the owner of the secondary mod to be responsible for duplicating original tuxemon assets into their mod's folder? Any drawbacks other than bloating the overall filesize of Tuxemon?

**Comment by bitcraft** [comment]
To clarify, we can gracefully handle sharing, sounds, music, sprites and other graphics between mods.  The issue is that we cannot share tilesets or images from maps.  So maps will need to be duplicated **right now**.  We can modify the map loaders to transform paths before resources are loaded and then we can also share maps (and their content).

**Comment by xirsoi** [comment]
Ah, and in my experience that is a relatively straightforward change to make.

**Comment by bitcraft** [comment]
Yeah, should not be too hard.  I will take this one actually, since I will be working on the map loader as part of the client view changes.

**Comment by bitcraft** [comment]
verified that it is possible to share maps if the map XML is manually edited to make the resources relative to the mods folder, rather than the maps folder where the map lives.
